nosnoc/solver.py

```diff
@@ -433,8 +433,8 @@ def solve(self) -> dict:
             if not check_ipopt_success(status):
                 print(f"Warning: IPOPT exited with status {status}")
 
-            if complementarity_residual < opts.comp_tol:
```

Comment:
This is the check for complementarity satisfaction, it shouldn't be commented out.